### PR TITLE
add test for 'PerformanceObserverInit.buffered' 

### PR DESCRIPTION
**Pull request by cvazac**
per proposed spec update:
https://github.com/w3c/performance-timeline/pull/76

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/229277441)

# Firefox (nightly channel) #
Testing web-platform-tests at revision c6ac532d627de0ae250eab3a7e2b4f77fbf699a3
Using browser at version BuildID 20170504100357; SourceStamp 33b92d9c40562dab3d7b602368c75619f1d793f7
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5841/performance-timeline/po-observe.html">/performance-timeline/po-observe.html</a></summary>


|                                    Subtest                                     | Results |                              Messages                              |
|--------------------------------------------------------------------------------|---------|--------------------------------------------------------------------|
|                                                                                | OK      |                                                                    |
| `PerformanceObserverInit.buffered should retrieve previously buffered entries` | FAIL    | `The expression cannot be converted to return the specified type.` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/229277453)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 77a6f944471ef975e39631f5636d8dbdd31f2612

**Comment by igrigorik** [comment]
This looks great! 

Would it make sense to also add another test that verifies that multiple entries are queued when observing different types with buffered data? E.g. navigation and user timing entries with buffered flag set to true?

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/230529089)

# Firefox (nightly) #
Testing web-platform-tests at revision 7a133ccc24944540cff6e732e6c34faedf416547
Using browser at version BuildID 20170509100208; SourceStamp b21b974d60d3075ae24f6fb1bae75d0f122f28fc
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5841/performance-timeline/po-observe.html">/performance-timeline/po-observe.html</a></summary>


|                                    Subtest                                     | Results |     Messages     |
|--------------------------------------------------------------------------------|---------|------------------|
|                                                                                | TIMEOUT |                  |
| `PerformanceObserverInit.buffered should retrieve previously buffered entries` | TIMEOUT | `Test timed out` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/230529099)

# Chrome (unstable) #
Testing web-platform-tests at revision a776a99d502fa8a4b5efda624bc8ea4122d0eadd
Using browser at version 60.0.3088.3 dev
Starting 10 test iterations

File: performance-timeline/po-observe.html
Comment (plehegar) [inline performance-timeline/po-observe.html]: do you actually use anything from that script? If not, I suggest removing the reference.

Comment (cvazac) [inline performance-timeline/po-observe.html]: yes, I use it for `checkEntries()`

File: performance-timeline/po-observe.html
Comment (yoavweiss) [inline performance-timeline/po-observe.html]: If I'm reading this correctly, when the test will fail it will timeout rather than provide useful output. Can you add an onload handler, which shouldn't be reached normally, prints out a failure message and calls `t.done()`?

**Review by yoavweiss — CHANGES_REQUESTED** [review]
LGTM! One nit to address though

**Review by igrigorik — APPROVED** [review]
lgtm, modulo Yoav's suggestion on timeout.

**Comment by igrigorik** [comment]
@cvazac build is failing: https://travis-ci.org/w3c/web-platform-tests/jobs/231258260#L349

@plehegar could you take another pass before we merge? :)

**Review by yoavweiss — APPROVED** [review]
LGTM

File: performance-timeline/po-observe.html
Comment (plehegar) [inline performance-timeline/po-observe.html]: I checked with @jgraham and this is will likely to make the test unstable in CIs. It seems better to let the test timeout.

**Comment by yoavweiss** [comment]
@plehegar PTAL?

**Review by plehegar — APPROVED** [review]

**Comment by plehegar** [comment]
This is good to be merged imho.

**Comment by igrigorik** [comment]
Done. Thanks everyone!

**Pull request merged**
